src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -86,7 +94,7 @@ public static Checksum Create(WellKnownSynchronizationKind kind, IObjectWritable
         {
             using var stream = SerializableBytes.CreateWritableStream();
 
-            using (var objectWriter = new ObjectWriter(stream, leaveOpen: true))
+            using (var objectWriter = new ObjectWriter(stream, leaveOpen: true, keepAliveCallback: ChecksumKeepAliveCallback))
```

Comment:
i'm skeptical this would ahve an effect. t agging @jasonmalinowski .

because GC.keepalive doesn't root something.  it just ensures that within teh scope of a method that the object isn't released.  i.e. if i have a scope like:

```
{
      object o  = ...
      DoSomething();
      DoSomething();
      GC.keepalive(o);
}
```

this just effectively keeps 'o' alive until that line.  after that line it can still be cleaned up.

also, since this is just for checksum computation, i don't get the need for keepalive semantics at all.   

(something just isn't jiving here).

Reply:
> also, since this is just for checksum computation, i don't get the need for keepalive semantics at all.

It's required to prevent a change in behavior relative to the original code.

Reply:
> i'm skeptical this would ahve an effect.

The caller might expect a call to `KeepAlive` to behave _at minimum_ like a call to `GC.KeepAlive`. Since no `ObjectReader` is ever created by this stream, only the minimum keep alive support is required. This is already documented on the property, and is the reason why the delegate was abstracted to a property.

> (something just isn't jiving here)

The code is probably wrong, but the only impact of this pull request is it revealed that the current approach is likely not what we want.

Reply:
➡️ Filed #46166